tests/javax/microedition/location/TestLocation.java

```diff
@@ -0,0 +1,50 @@
+package javax.microedition.location;
+
+import gnu.testlet.TestHarness;
+import gnu.testlet.Testlet;
+import gnu.testlet.TestUtils;
+import com.nokia.mid.location.LocationUtil;
+
+public class TestLocation implements Testlet, LocationListener {
+    TestHarness th;
+
+    public void locationUpdated(LocationProvider provider, final Location location) {
+        try {
+            QualifiedCoordinates c = location.getQualifiedCoordinates();
+            if (c != null) {
+                System.out.println("new Location Latitude: " + c.getLatitude() +
+                                   " Logitude: " + c.getLongitude());
+
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            th.fail("Unexpected exception: " + e);
+        }
+        // TODO There is an unknown bug that causes the following line blocking
+        // the VM.
+        // provider.setLocationListener(null, -1, -1, -1);
```

Comment:
The VM get stuck if this line is enabled. Might be something related to thread scheduler. I will take a look tomorrow.